src/intel.cpp

```diff
@@ -7,64 +7,51 @@ using json = nlohmann::json;
 
 static bool init_intel = false;
 struct gpuInfo gpu_info_intel {};
+static char drm_dev[10];
 
 void intelGpuThread(){
     init_intel = true;
-    static char stdout_buffer[1024];
-    FILE* intel_gpu_top = popen("intel_gpu_top -J -s 500", "r");
-    int num_line = 0;
-    std::string buf;
-    int num_iterations = 0;
-    while (fgets(stdout_buffer, sizeof(stdout_buffer), intel_gpu_top)) {
-        if (num_line > 0)
-            buf += stdout_buffer;
-
-        num_line++;
-        if (strlen(stdout_buffer) < 4 && !strchr(stdout_buffer, '{') && !strchr(stdout_buffer, ',')) {
-            if (buf[0] != '{')
-                buf = "{\n" + buf;
-
-            if (num_iterations > 0){
-                buf += "\n}";
-                json j = json::parse(buf);
-                if  (j.contains("engines"))
-                    if (j["engines"].contains("Render/3D/0"))
-                        if (j["engines"]["Render/3D/0"].contains("busy"))
-                            gpu_info_intel.load = j["engines"]["Render/3D/0"]["busy"].get<int>();
+    static char stdout_buffer[4096];
 
-                if (j.contains("frequency"))
-                    if (j["frequency"].contains("actual"))
-                        gpu_info_intel.CoreClock = j["frequency"]["actual"].get<int>();
-                if (j.contains("power")){
-                    if (j["power"].contains("GPU"))
-                        gpu_info_intel.powerUsage = j["power"]["GPU"].get<float>();
-                    if (j["power"].contains("Package"))
-                        gpu_info_intel.apu_cpu_power = j["power"]["Package"].get<float>();
-                }
-
-            }
-            buf = "";
-            num_line = 0;
+    std::string cmd("/home/kk/local-data/projects/upstreams/MangoHud/build/src/mango_intel_stats ");
+    cmd += drm_dev;
+    FILE* intel_gpu_top = popen(cmd.c_str() , "r");
+    while (fgets(stdout_buffer, sizeof(stdout_buffer), intel_gpu_top)) {
+        if ( stdout_buffer[0] != '{' || stdout_buffer[strlen(stdout_buffer)-1] != '\n') {
+            SPDLOG_ERROR("Overran 4k buffer for fgets output. Expect sadness:\n {}", stdout_buffer);
+            continue;
         }
-        num_iterations++;
+
+        json j = json::parse(stdout_buffer);
+        if  (j.contains("gpu_busy_pct"))
+            gpu_info_intel.load = j["gpu_busy_pct"].get<int>();
+        if  (j.contains("gpu_clock_mhz"))
+            gpu_info_intel.CoreClock = j["gpu_clock_mhz"].get<int>();
+        if  (j.contains("dgpu_power_w"))
+            gpu_info_intel.powerUsage = j["dgpu_power_w"].get<int>();
+        if  (j.contains("cpu_power_w"))
+            gpu_info_intel.apu_cpu_power = j["cpu_power_w"].get<int>();
     }
 
     int exitcode = pclose(intel_gpu_top) / 256;
     if (exitcode > 0){
         if (exitcode == 127)
-        SPDLOG_INFO("Failed to open '{}'", "intel_gpu_top");
+        SPDLOG_INFO("Failed to open '{}'", "intel_perf_stats");
```

Comment:
Should these error messages say `mango_intel_stats` instead of `intel_perf_stats`?

Reply:
Ah yea, I went back and forth on the name a bit. Thanks.